infrastructures/kubernetes/src/main/java/org/eclipse/che/workspace/infrastructure/kubernetes/provision/KubernetesCheApiInternalEnvVarProvider.java

```diff
@@ -29,12 +30,15 @@
 
   @Inject
   public KubernetesCheApiInternalEnvVarProvider(
-      @Named("che.api.internal") String cheServerEndpoint) {
+      @Nullable @Named("che.api.internal") String cheServerEndpoint) {
     this.cheServerEndpoint = cheServerEndpoint;
   }
 
   @Override
   public Pair<String, String> get(RuntimeIdentity runtimeIdentity) throws InfrastructureException {
-    return Pair.of(CHE_API_INTERNAL_VARIABLE, cheServerEndpoint);
+    if (this.cheServerEndpoint != null) {
+      return Pair.of(CHE_API_INTERNAL_VARIABLE, cheServerEndpoint);
+    }
+    return null;
```

Comment:
This breaks https://github.com/eclipse-che/che-server/blob/07263f1e30089689d71b057f747a44a29283e3c4/wsmaster/che-core-api-workspace/src/main/java/org/eclipse/che/api/workspace/server/spi/provision/env/CheApiEnvVarProvider.java#L48

while that env var is deprecated, we are still providing it to the workspace pods as far as I am aware.

Reply:
CHE_API_VARIABLE !=che.api.internal @metlos can you explain how it's related?

Reply:
The current behavior is that che server endpoint is never null and the `get` method of the `KubernetesCheApiInternalEnvVarProvider` never returns a `null` value. This is the assumption in `CheApiEnvVarProvider` which uses the return value from this method without a null check.

By making the `KubernetesCheApiInternalEnvVarProvider` potentially return `null`, we're breaking that assumption and open the possibility of an NPE. I have no idea if that is possible practically, just see the possibility in the code.

Reply:
I think the whole point of this pr is that internal can be null. If some code needs something that never can be null it should use the results of CheApiEnvVarProvider.java right?

Reply:
+1 for @skabashnyuk conclusions.

Reply:
And I think we have null check https://github.com/eclipse-che/che-server/pull/45/files#diff-a433d052bb9fbe17e92d0cac528935ba2dfae6c982ce08dcd95c0b23b26403a1R52. I see CheApiEnvVarProvider is deprecated. So ideally not null value api endpoint should provide CheApiExternalEnvVarProvider like successor. This provider should provide public url. So logic is simple for application which used api env variables: if internal url env variable is absent or empty, than should be used value from external public url env variable. See pr: https://github.com/eclipse-che/che-theia/pull/1161. Also for che-theia I removed usage deprecated CheApiEnvVarProvider in this pr.

Reply:
That null-check was what I was after. Thanks for adding it :)